Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl

```diff
@@ -0,0 +1,437 @@
+    sharedLength[indexForSharedMem] = g_HairRestLengthSRV[globalVertexIndex] * CM_TO_METERS;
+
+    GroupMemoryBarrierWithGroupSync();
+
+/*
+    //------------
+    // Wind
+    //------------
+    if ( g_Wind.x != 0 || g_Wind.y != 0 || g_Wind.z != 0 )
+    {
+        float4 force = float4(0, 0, 0, 0);
+
+        float frame = g_Wind.w;
+
+        if ( localVertexIndex >= 2 && localVertexIndex < numVerticesInTheStrand-1 )
+        {
+            // combining four winds.
+            float a = ((float)(globalStrandIndex % 20))/20.0f;
+            float3  w =  a* g_Wind.xyz + (1.0f - a) * g_Wind1.xyz + a * g_Wind2.xyz + (1.0f - a) * g_Wind3.xyz;
+//            float3  w = float3(5.2, 0, 0);
+
+            uint sharedIndex = localVertexIndex * numOfStrandsPerThreadGroup + localStrandIndex;
+
+            float3 v = sharedPos[sharedIndex].xyz - sharedPos[sharedIndex+numOfStrandsPerThreadGroup].xyz;
+            float3 force = -cross(cross(v, w), v);
+            sharedPos[sharedIndex].xyz += force*g_TimeStep*g_TimeStep;
+        }
+    }
+
+    GroupMemoryBarrierWithGroupSync();
+*/
+    //----------------------------
+    // Enforce length constraints
+    //----------------------------
+    uint a = floor(numVerticesInTheStrand/2.0f);
+    uint b = floor((numVerticesInTheStrand-1)/2.0f);
+
+    int nLengthContraintIterations = GetLengthConstraintIterations();
+
+    for ( int iterationE=0; iterationE < nLengthContraintIterations; iterationE++ )
+    {
+        uint sharedIndex = 2*localVertexIndex * numOfStrandsPerThreadGroup + localStrandIndex;
+
+        if( localVertexIndex < a )
+            ApplyDistanceConstraint(sharedPos[sharedIndex], sharedPos[sharedIndex+numOfStrandsPerThreadGroup], sharedLength[sharedIndex].x);
+
+        GroupMemoryBarrierWithGroupSync();
+
+        if( localVertexIndex < b )
+            ApplyDistanceConstraint(sharedPos[sharedIndex+numOfStrandsPerThreadGroup], sharedPos[sharedIndex+numOfStrandsPerThreadGroup*2], sharedLength[sharedIndex+numOfStrandsPerThreadGroup].x);
+
+        GroupMemoryBarrierWithGroupSync();
+    }
+
+    //------------------------------------------
+    // Collision handling with capsule objects
+    //------------------------------------------
+    float4 oldPos = g_HairVertexPositionsPrev[globalVertexIndex];
+    bool bAnyColDetected = false;       // Adi
+//    bool bAnyColDetected = ResolveCapsuleCollisions(sharedPos[indexForSharedMem], oldPos);
+    GroupMemoryBarrierWithGroupSync();
+
+    //-------------------
+    // Compute tangent
+    //-------------------
+    // If this is the last vertex in the strand, we can't get tangent from subtracting from the next vertex, need to use last vertex to current
+    uint indexForTangent = (localVertexIndex == numVerticesInTheStrand - 1) ? indexForSharedMem - numOfStrandsPerThreadGroup : indexForSharedMem;
+    float3 tangent = sharedPos[indexForTangent + numOfStrandsPerThreadGroup].xyz - sharedPos[indexForTangent].xyz;
```

Comment:
original TressFX and while you are right, this will dictate having edge case handling.  Having such high resolution for the hair makes this point less important or visible and the critical point becomes about having more control point when creating the hair (since a strand will have ~32 length segments, having more control point during the creation will take better use of them).